.github/actions/dev-env-setup/action.yml

```
@@ -0,0 +1,56 @@
+name: "Set up registration dev environment"
+description: "Sets up asdf and configures the cache"
+runs:
+  using: composite
+  steps:
+    - name: asdf setup
+      uses: asdf-vm/actions/setup@v1
+    - uses: actions/cache@v3
+      id: asdf-cache-client
+      with:
+        path: |
+          ~/.asdf
+          ./client/.tool-versions
+        key: ${{ runner.os }}-asdf-cache-client-${{ hashFiles('client/.tool-versions') }}
+    - uses: actions/cache@v3
+      id: asdf-cache-backend
+      with:
+        path: |
+          ~/.asdf
+          ./bc_obps/.tool-versions
+        key: ${{ runner.os }}-asdf-cache-backend-${{ hashFiles('bc_obps/.tool-versions') }}
+    - uses: actions/cache@v3
+      id: yarn-cache
+      with:
+        path: |
+          ~/.cache/yarn
+          ./client/node_modules
+        key: ${{ runner.os }}-yarn-cache-${{ hashFiles('client/yarn.lock') }}-v2
+    - name: Set up python
+      id: setup-python
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.9.16"
```

Probably enough to specify the minor to make sure we get the updates without work on our end
```suggestion
        python-version: "3.9"
```